### PR TITLE
Add cctag/1.0.3

### PR DESCRIPTION
**Pull request by EstebanDugueperoux2**
Specify library name and version:  **cctag/1.0.3**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by EstebanDugueperoux2** [comment]
Depends of https://github.com/conan-io/conan-center-index/pull/18625 to have pipeline v2 works.

**Comment by EstebanDugueperoux2** [comment]
Hi @ericLemanissier , @prince-chrismc, @SpaceIm could you review this PR?

**Review by ericLemanissier — APPROVED** [review]

File: recipes/cctag/all/patches/0001-Ignore-TBB-version.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+@@ -241,7 +241,7 @@ endif()
+ # FIND Intel TBB
+ # With MVSC, CMAKE_BUILD_TYPE will always be None, so TBB_USE_DEBUG_BUILD will always be false.
+ string(COMPARE EQUAL "${CMAKE_BUILD_TYPE}" Debug TBB_USE_DEBUG_BUILD)
+-find_package(TBB 2021.5.0 CONFIG REQUIRED)
++find_package(TBB CONFIG REQUIRED)
+ message(STATUS "Found TBB: version ${TBB_VERSION}")
+ 
+ add_subdirectory(src)
+
```
Comment (ericLemanissier) [inline recipes/cctag/all/patches/0001-Ignore-TBB-version.patch:12]:
```suggestion
```

File: recipes/cctag/all/conanfile.py [hunk 2]
```diff
@@ -3,7 +3,8 @@
 from conan.tools.build import check_min_cppstd
 from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
 from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rmdir
-from conan.tools.microsoft import is_msvc_static_runtime
+from conan.tools.microsoft import is_msvc_static_runtime, is_msvc
+from conan.tools.scm import Version
```
Comment (ericLemanissier) [inline recipes/cctag/all/conanfile.py:7]:
```suggestion
```

File: recipes/cctag/all/conandata.yml [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+    url: "https://github.com/alicevision/CCTag/archive/refs/tags/v1.0.3.tar.gz"
+    sha256: "25396b03c4aa3c1be241a2a8518a29511cb4fff695caa5126203fbba8e8e444d"
   "1.0.1":
     url: "https://github.com/alicevision/CCTag/archive/refs/tags/v1.0.1.tar.gz"
     sha256: "ae8a819bc978eb13bb1061a204c214da835e56c9b7dc775237ed6b2191011dec"
 patches:
+  "1.0.3":
+    - patch_file: "patches/0001-Ignore-TBB-version.patch"
+      patch_description: "Ignore TBB version in find_package to consume one from conan"
+      patch_type: "conan"
```
Comment (ericLemanissier) [inline recipes/cctag/all/conandata.yml:12]:
```suggestion
```

**Review by ericLemanissier — COMMENTED** [review]
This patch does not seem needed, and could make the situation more confusing than needed if somehow the version of TBB is overriden duing the build of cctag

**Comment by EstebanDugueperoux2** [comment]
> This patch does not seem needed, and could make the situation more confusing than needed if somehow the version of TBB is overriden duing the build of cctag

Indeed but the issue is onetbb/2021.5.0 is not available on conan center.
Then I must before add it to conan center to reference it in this recipe, and perhaps add a validation rule to forbidden the onetbb/2021.5.0 requirement override.

**Comment by ericLemanissier** [comment]
ow, so cmake does not consider 2021.6.0 compatible with 2021.5.0 ? ok then

**Comment by EstebanDugueperoux2** [comment]
> ow, so cmake does not consider 2021.6.0 compatible with 2021.5.0 ? ok then

It seems that TBB does not follow semantic versionning.

**Comment by EstebanDugueperoux2** [comment]
Hi @jcar87 , hi @prince-chrismc do you have an idea of what occurred in my pipeline?

**Pull request closed without merging**